File: src/test/java/net/igsoft/sdi/testclasses/DCreator.java
package net.igsoft.sdi.testclasses;

import java.util.List;

import com.google.common.collect.Lists;

import net.igsoft.sdi.creator.CreatorBase;
import net.igsoft.sdi.engine.InstanceProvider;
import net.igsoft.sdi.parameter.LaunchType;

public class DCreator extends CreatorBase<DClass, LaunchType> {
    @Override
    public DClass create(InstanceProvider instanceProvider, LaunchType launchType) {
        EClass e = instanceProvider.getOrCreate(EClass.class, launchType);
        return new DClass(e, instanceProvider.getOrCreate(Stepper.class));
    }

    @Override
    public List<CreatorBase<?, ?>> defaultCreators() {
        return Lists.newArrayList(new ECreator());
    }
}

File: src/test/scala/net/igsoft/sdi/scala/testclasses/DCreator.scala
package net.igsoft.sdi.scala.testclasses

import net.igsoft.sdi.{Creator, InstanceCreator, LaunchType}

class DCreator extends Creator[D, LaunchType] {
  override def create(instanceCreator: InstanceCreator, launchType: LaunchType): D = {
    val e = instanceCreator.getOrCreate(classOf[E], launchType)
    new D(e, instanceCreator.getOrCreate(classOf[Stepper]))
  }
}
